cmd/xl-v1-common.go

```diff
@@ -58,14 +58,14 @@ func (xl xlObjects) isObject(bucket, prefix string) (ok bool) {
 
 	g := errgroup.WithNErrs(len(storageDisks))
 
-	for index, disk := range storageDisks {
-		if disk == nil {
-			continue
-		}
-		index := index
+	for index := range storageDisks {
+		disk := storageDisks[index]
```

Comment:
you should make sure the `index` is re-assigned as done previously. and let the `storageDisks[index]` stay as is meaning the right fix is 

```
for index := range storageDisks {
     index := index 
     g.Go(func() error {
            if storageDisks[index] == nil {
                   return errDiskNotFound
            }
     ....
```

Reply:
@harshavardhana it is the same thing I am doing with 'disk' variable, disk does not come from range storageDisks but it is re-initialized in each loop

Reply:
> @harshavardhana it is the same thing I am doing with 'disk' variable, disk does not come from range storageDisks but it is re-initialized in each loop

But maybe your way is more clear, I will do it and add a comment

Reply:
https://play.golang.org/p/QpwMC0xsrBA